src/coreclr/tools/ILVerification/ILImporter.Verify.cs

```diff
@@ -2647,6 +2648,18 @@ void CheckPendingPrefix(Prefix mask)
             Check((mask & Prefix.Constrained) == 0, VerifierError.Constrained);
         }
 
+        static bool HasIsExternalInit(MethodSignature signature)
+        {
+            if (signature.HasEmbeddedSignatureData)
+                foreach (var data in signature.GetEmbeddedSignatureData())
+                {
+                    if (data.type is MetadataType mdType && mdType.Namespace == "System.Runtime.CompilerServices" && mdType.Name == "IsExternalInit")
```

Comment:
According to https://github.com/dotnet/runtime/issues/34978, this should be only recognized on return type. This code will recognize it anywhere in the signature.

I think this should also check for IndexOfCustomModifiersOnReturnType

Reply:
Added `data.index == IndexOfCustomModifiersOnReturnType` check.